transportable-udfs-annotation-processor/src/main/java/com/linkedin/transport/processor/TransportProcessor.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright 2019 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.processor;
+
+import com.linkedin.transport.api.udf.StdUDF;
+import com.linkedin.transport.api.udf.TopLevelStdUDF;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.LinkedList;
+import java.util.Set;
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Filer;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.annotation.processing.SupportedOptions;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+import javax.tools.Diagnostic;
+import javax.tools.FileObject;
+import javax.tools.StandardLocation;
+
+
+/**
+ * Annotation Processor for Transport UDFs which extracts UDF metadata from user defined UDF classes and stores it in a
+ * resource file
+ *
+ * The annotation processor does not rely on a special annotation, instead it will look for non-abstract classes which
+ * extend {@link StdUDF} and implement {@link TopLevelStdUDF}. It will then identify all overloadings of the same UDF
+ * name and store them in the resource file.
+ */
+@SupportedOptions({"debug"})
+@SupportedAnnotationTypes({"*"})
+public class TransportProcessor extends AbstractProcessor {
+
+  private Types _types;
+  private Elements _elements;
+  private TypeMirror _topLevelStdUDFInterfaceType;
+  private TypeMirror _stdUDFClassType;
+  private UDFProperties _udfProperties;
+
+  @Override
+  public SourceVersion getSupportedSourceVersion() {
+    return SourceVersion.latestSupported();
+  }
+
+  @Override
+  public synchronized void init(ProcessingEnvironment processingEnv) {
+    super.init(processingEnv);
+    _types = processingEnv.getTypeUtils();
+    _elements = processingEnv.getElementUtils();
```

Comment:
Why is this a member variable?

Reply:
Previously it was not being used outside init() but now it is.